### PR TITLE
breakout the component types into a scoped enum

### PR DESCRIPTION
**Pull request by aronwk-aaron**
tested that things are the same as they were before

### CODE REVIEW

File: dDatabase/Tables/CDComponentsRegistryTable.h [hunk 1]
```diff
@@ -7,11 +7,11 @@
   \file CDComponentsRegistryTable.hpp
```
Comment (EmosewaMC) [inline dDatabase/Tables/CDComponentsRegistryTable.h:7]: lol.  hpp.

File: dGame/dComponents/RebuildComponent.h [hunk 2]
```diff
@@ -19,7 +20,7 @@ class Entity;
  */
 class RebuildComponent : public Component {
 public:
-	static const uint32_t ComponentType = COMPONENT_TYPE_REBUILD;
+	static const eReplicaComponentType ComponentType = eReplicaComponentType::REBUILD;
```
Comment (EmosewaMC) [inline dGame/dComponents/RebuildComponent.h:23]: Rebuild or quickbuild?  I've seen both in this PR and getting this translation correct is critical

File: dGame/dMission/MissionTask.cpp [hunk 3]
```diff
@@ -237,7 +238,7 @@ void MissionTask::Progress(int32_t value, LWOOBJID associate, const std::string&
 			break;
 		}
 
-		activity = static_cast<ScriptedActivityComponent*>(entity->GetComponent(COMPONENT_TYPE_REBUILD));
+		activity = static_cast<ScriptedActivityComponent*>(entity->GetComponent(eReplicaComponentType::QUICK_BUILD));
```
Comment (EmosewaMC) [inline dGame/dMission/MissionTask.cpp:241]: Rebuild

File: dGame/dGameMessages/GameMessages.cpp [hunk 4]
```diff
@@ -4968,7 +4969,7 @@ void GameMessages::HandleRebuildCancel(RakNet::BitStream* inStream, Entity* enti
 	inStream->Read(bEarlyRelease);
 	inStream->Read(userID);
 
-	RebuildComponent* rebComp = static_cast<RebuildComponent*>(entity->GetComponent(COMPONENT_TYPE_REBUILD));
+	RebuildComponent* rebComp = static_cast<RebuildComponent*>(entity->GetComponent(eReplicaComponentType::QUICK_BUILD));
```
Comment (EmosewaMC) [inline dGame/dGameMessages/GameMessages.cpp:4972]: rebuild

File: dGame/Entity.cpp [hunk 5]
```diff
@@ -1087,99 +1088,99 @@ void Entity::WriteComponents(RakNet::BitStream* outBitStream, eReplicaPacketType
...
 
 	BaseCombatAIComponent* baseCombatAiComponent;
-	if (TryGetComponent(COMPONENT_TYPE_BASE_COMBAT_AI, baseCombatAiComponent)) {
+	if (TryGetComponent(eReplicaComponentType::BASE_COMBAT_AI, baseCombatAiComponent)) {
 		baseCombatAiComponent->Serialize(outBitStream, bIsInitialUpdate, flags);
 	}
 
 	RebuildComponent* rebuildComponent;
-	if (TryGetComponent(COMPONENT_TYPE_REBUILD, rebuildComponent)) {
+	if (TryGetComponent(eReplicaComponentType::QUICK_BUILD, rebuildComponent)) {
```
Comment (EmosewaMC) [inline dGame/Entity.cpp:1116]: rebuild

File: dGame/Entity.cpp [hunk 6]
```diff
@@ -497,24 +498,24 @@ void Entity::Initialize() {
...
 		BaseCombatAIComponent* comp = new BaseCombatAIComponent(this, combatAiId);
-		m_Components.insert(std::make_pair(COMPONENT_TYPE_BASE_COMBAT_AI, comp));
+		m_Components.insert(std::make_pair(eReplicaComponentType::BASE_COMBAT_AI, comp));
 	}
 
-	if (int componentID = compRegistryTable->GetByIDAndType(m_TemplateID, COMPONENT_TYPE_REBUILD) > 0) {
+	if (int componentID = compRegistryTable->GetByIDAndType(m_TemplateID, eReplicaComponentType::QUICK_BUILD) > 0) {
 		RebuildComponent* comp = new RebuildComponent(this);
-		m_Components.insert(std::make_pair(COMPONENT_TYPE_REBUILD, comp));
+		m_Components.insert(std::make_pair(eReplicaComponentType::QUICK_BUILD, comp));
```
Comment (EmosewaMC) [inline dGame/Entity.cpp:518]: rebuild

File: dGame/Entity.cpp [hunk 7]
```diff
@@ -268,61 +269,61 @@ void Entity::Initialize() {
...
 	}
 
 	/**
 	 * Multiple components require the destructible component.
 	 */
 
-	int buffComponentID = compRegistryTable->GetByIDAndType(m_TemplateID, COMPONENT_TYPE_BUFF);
-	int rebuildComponentID = compRegistryTable->GetByIDAndType(m_TemplateID, COMPONENT_TYPE_REBUILD);
+	int buffComponentID = compRegistryTable->GetByIDAndType(m_TemplateID, eReplicaComponentType::BUFF);
+	int rebuildComponentID = compRegistryTable->GetByIDAndType(m_TemplateID, eReplicaComponentType::QUICK_BUILD);
```
Comment (EmosewaMC) [inline dGame/Entity.cpp:326]: rebuild

**Review by EmosewaMC — CHANGES_REQUESTED** [review]
Found some mis-matched changes i'd personally like to see left out of the pr unless you plan on changing all the rebuilds to quick_build or vice versa

File: dGame/dComponents/RebuildComponent.h (hunk 2 above)
Comment (aronwk-aaron) [reply dGame/dComponents/RebuildComponent.h:23]: DLU rebuild == actual quick_build

**Review by EmosewaMC — APPROVED** [review]
I don't see any further missed changes

**Review by Wincent01 — APPROVED** [review]

**Pull request merged**